sphinx/texinputs/sphinxlatexobjects.sty

```diff
@@ -167,6 +183,76 @@
   \nopagebreak\noindent\kern-\labelwidth\sphinxcode{)}{#3}
   \pysigadjustitemsep
 }
+\newcommand{\pysigwithonelineperargwithonelinepertparg}[4]{
+  % #1 = name, #2 = tplist, #3 = arglist, #4 = retann
+  % render each type parameters and argument lists on its own line
+  \item[#1\sphinxcode{[}\strut]
+  \leavevmode\par\nopagebreak
+  \begingroup
+     \let\sphinxparamcomma\sphinxparamcommaoneperline
+     % \sphinxtypeparam is treated similarly to \sphinxparam but since
+     % \sphinxoptional is not accepted in a type parameters list, we do
+     % not need the hook or the global definition
+     \let\spx@sphinxtypeparam\sphinxtypeparam
+     \def\sphinxtypeparam{\def\sphinxtypeparam{\par\spx@sphinxtypeparam}\spx@sphinxtypeparam}%
+  #2\par
+  \endgroup
+  \nopagebreak\noindent\kern-\labelwidth\sphinxcode{]}
+  % render the rest of the signature like in \pysigwithonelineperarg
+  \sphinxcode{(}\strut\leavevmode\par\nopagebreak
```

Comment:
Thank you for the explanation ! by the way, I see now that there is a thin space between `]` and `(` in the last two cases since I forgot to had `%` at the end of my line.